### PR TITLE
PEPPER-1561 Enable GA for lms & Osteo

### PR DESCRIPTION
**Pull request by ssettipalli**
PEPPER-1561
Enable front-end static pages GA for Osteo and LMS
Make sure once user is logged in , NO GA calls in network

Assigned GA tag: G-VZ3QPCL198 for non-prod envs for testing/QA

Steps to try locally
render pepperConfig.js
./build-study.sh [version] [env] . osteo CMI-OSTEO --config
./build-study.sh [version] [env] . lms cmi-osteo --config

verify generated output-config/pepperConfig.js

bring up study UI locally with backend pointing to Dev
http://localhost:4200/
click on static pages , do a signup, login, submit surveys
Login into https://analytics.google.com/analytics , pepper-demo-site to verify GA data
GA should log only static pages and NO other GA calls  in devtools network

DEV QA
Click on several static pages, 
Do new signup, submit surveys, activity page refreshes 
Do login , page refreshes.. so on

Login into https://analytics.google.com/analytics , pepper-demo-site to verify GA data
GA should log only static pages and NO other GA calls  in devtools network
Both studies use same GA Tag (pepper-demo-site )

Pending:
Will be doing additional testing in Dev post merge.

### CODE REVIEW

File: config/pepperConfig.js.ctmpl [hunk 1]
```diff
@@ -142,45 +144,6 @@ var DDP_ENV = {
     {{end}}
 {{end}}
 
-{{if eq $study_guid "testboston"}}
```
Comment (ssettipalli) [inline config/pepperConfig.js.ctmpl:145]: additional non active studies related cleanup

File: config/pepperConfig.js.ctmpl [hunk 2]
```diff
@@ -113,14 +113,16 @@ var DDP_ENV = {
...
+        {{ else if eq $study_guid "cmi-lms"}}
+            DDP_ENV['projectGAToken'] = "G-KB65C93K2D";
         {{ else if eq $study_guid "cmi-pancan"}}
             DDP_ENV['projectGAToken'] = "UA-64885513-10";
         {{ else if eq $study_guid "circadia"}}
             DDP_ENV['projectGAToken'] = "UA-207512065-1";
         {{end}}
     {{else}}
-        DDP_ENV['projectGAToken'] = "UA-108484823-1";
+        DDP_ENV['projectGAToken'] = "G-VZ3QPCL198";
```
Comment (ssettipalli) [inline config/pepperConfig.js.ctmpl:125]: non-prod GA tag for both studies

File: ddp-workspace/projects/ddp-lms/src/app/app.module.ts [hunk 3]
```diff
@@ -191,4 +194,23 @@ const translateFactory =
   ],
   bootstrap: [AppComponent],
 })
-export class AppModule {}
+
+export class AppModule {
```
Comment (ssettipalli) [inline ddp-workspace/projects/ddp-lms/src/app/app.module.ts:197]: GA Tag is loaded from rendered config file.

File: ddp-workspace/projects/ddp-lms/src/app/app.module.ts [hunk 4]
```diff
@@ -191,4 +194,23 @@ const translateFactory =
...
+  constructor(
+    private router: Router) {
+    this.router.events.subscribe(event => {
+      if (event instanceof NavigationEnd) {
+        if (event.url !== 'undefined' &&
```
Comment (ssettipalli) [inline ddp-workspace/projects/ddp-lms/src/app/app.module.ts:207]: Make gtag calls for only these specified static pages. 
If there is some easy way to check if user is logged-in, we could use this but this works and verified locally that NO network calls from anything else other than these static pages

File: ddp-workspace/projects/ddp-lms/src/app/components/user-activities/user-activities.component.ts [hunk 5]
```diff
@@ -136,6 +136,7 @@ export class UserActivitiesComponent implements OnInit, OnDestroy, OnChanges, Af
   }
 
   private doAnalytics(action: string): void {
-      this.analytics.emitCustomEvent(AnalyticsEventCategories.Dashboard, action);
+    //skip user activity events
+    //this.analytics.emitCustomEvent(AnalyticsEventCategories.Dashboard, action);
```
Comment (ssettipalli) [inline ddp-workspace/projects/ddp-lms/src/app/components/user-activities/user-activities.component.ts:140]: commented out old GA calls for extra safety

File: ddp-workspace/projects/ddp-lms/src/app/app.module.ts (hunk 4 above)
Comment (zyme) [reply ddp-workspace/projects/ddp-lms/src/app/app.module.ts:207]: Awesome!

File: ddp-workspace/projects/ddp-lms/src/index.html [hunk 6]
```diff
@@ -20,7 +20,15 @@
     <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet" />
 
     <script src="assets/config/pepperConfig.js"></script>
-  </head>
+
+    <script async src="https://www.googletagmanager.com/gtag/js?id=DDP_ENV.projectGAToken"></script>
+    <script>
+      window.dataLayer = window.dataLayer || [];
+      function gtag(){dataLayer.push(arguments);}
+      gtag('js', new Date());
```
Comment (zyme) [inline ddp-workspace/projects/ddp-lms/src/index.html]: Can we move this into app.module.ts?  I worry about loading it here globally.

Comment (ssettipalli) [reply ddp-workspace/projects/ddp-lms/src/index.html]: Coldn't get it work when I tried to move all statements. 
Moved gtag('js', new Date()); which seem to be fine.

Comment (ssettipalli) [reply ddp-workspace/projects/ddp-lms/src/index.html]: Was able to move the gtag initializing javascript from index.html to app.module.ts
Doesn't look clean but seem to work .

**Review by zyme — APPROVED** [review]

**Pull request merged**